pkg/extensions/customresources.go

```diff
@@ -256,7 +256,7 @@ func RestoreExtensionWithDeployFunction(
 ) error {
 	extensionObj, err := deployFunc(ctx, v1beta1constants.GardenerOperationWaitForState)
 	if err != nil {
-		return err
+		return gardencorev1beta1helper.NewErrorWithCodes(err, gardencorev1beta1helper.DeprecatedDetermineErrorCodes(err)...)
```

Comment:
This should stay as it was before, because this is deployed to the seed and no user-specific error should occur. Similarly for other deploy functions as well.